### PR TITLE
Adds test for uint datatype for image_shape and filter_shape in thean…

### PR DESCRIPTION
**Pull request by mronian**
…o.tensor.nnet.conv and also checks if datatype is in tensor.discrete_dtypes

### CODE REVIEW

File: theano/tensor/nnet/tests/test_conv.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 from nose.plugins.skip import SkipTest
 import numpy
-
+import ctypes
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_conv.py]: I don't like importing this. We don't do it anywhere else so this add an extra dependency.

Can you use numpy.asarray(3, dtype='uint8') or similar instead?

thanks

**Comment by mronian** [comment]
Sorry about that. I have updated the pull request.

**Comment by nouiz** [comment]
There is some failure in the tests:

```
======================================================================

ERROR: Tests for uint datatype in filter_shape

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/theano/tensor/nnet/tests/test_conv.py", line 169, in test_uint_filter_shape_datatype

    self.validate((3, 2, 3, 3), (2, 2, 3, ctypes.c_uint(3)), 'valid', verify_grad=False)

NameError: global name 'ctypes' is not defined

======================================================================

ERROR: Tests for uint datatype in image_shape.

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/theano/tensor/nnet/tests/test_conv.py", line 161, in test_uint_image_shape_datatype

    self.validate((2, 2, 3, ctypes.c_uint(3)), (3, 2, 3, 3), 'valid', verify_grad=False)

NameError: global name 'ctypes' is not defined

======================================================================

ERROR: Tests for uint datatype in filter_shape

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/theano/tensor/nnet/tests/test_conv.py", line 169, in test_uint_filter_shape_datatype

    self.validate((3, 2, 3, 3), (2, 2, 3, ctypes.c_uint(3)), 'valid', verify_grad=False)

NameError: global name 'ctypes' is not defined

======================================================================

ERROR: Tests for uint datatype in image_shape.

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/theano/tensor/nnet/tests/test_conv.py", line 161, in test_uint_image_shape_datatype

    self.validate((2, 2, 3, ctypes.c_uint(3)), (3, 2, 3, 3), 'valid', verify_grad=False)

NameError: global name 'ctypes' is not defined

----------------------------------------------------------------------
```

**Comment by mronian** [comment]
I did not commit correctly before. Please check the PR now. Thanks for being so patient.

**Comment by mronian** [comment]
I did not commit correctly before. Please check the PR now. Thanks for being so patient.

**Comment by nouiz** [comment]
thanks

**Pull request merged**